syntax = "proto3";

package agent;

// ── Request / Response ──────────────────────────────────────────────

message AgentSettings {
    string model_name   = 1;
    float  temperature  = 2;
    int32  max_tokens   = 3;
}

message GenerateRequest {
    string        user_message   = 1;
    string        system_prompt  = 2;
    string        context_json   = 3;
    AgentSettings agent_settings = 4;
}

message GenerateResponse {
    string text          = 1;
    string metadata_json = 2;
}

// ── Service ─────────────────────────────────────────────────────────

service AgentService {
    rpc Generate (GenerateRequest) returns (GenerateResponse);
}
